lib/ansi.nix

```diff
@@ -0,0 +1,59 @@
+# This exports a basic set of ANSI escape sequences to colorize terminal output.
+
+{ ... }:
+
+let
+  # Escape character. "\e" or similar encodings currently not supported by Nix.
+  esc = "";
+  reset = "${esc}[0m";
+  style = {
+    bold = "${esc}[1m";
+    italic = "${esc}[3m";
+  };
+  colorCode = {
+    black = 0;
+    red = 1;
+    green = 2;
+    yellow = 3;
+    blue = 4;
+    magenta = 5;
+    cyan = 6;
+    white = 7;
+  };
+  color = {
+    fg = builtins.mapAttrs (name: value: "${esc}[3${toString value}m") colorCode;
+    bg = builtins.mapAttrs (name: value: "${esc}[4${toString value}m") colorCode;
+  };
+
+  /* Stylizes a string with ANSI escape sequences.
+
+     Type: stylize :: [String] -> String -> String
+  */
+  stylize =
+    # List of ANSI style attributes.
+    styles:
+    # Text to encapsulate by style.
+    text:
+    let
+      ansiBefore = builtins.concatStringsSep "" styles;
+    in
+    if text == ""
+    then ""
+    else "${ansiBefore}${text}${reset}";
```

Comment:
I am not sure if the intended behavior to just concat the empty string would be what we want here

Reply:
Not sure what you mean. There is no concatenation of empty strings.